### PR TITLE
Add dilation for exconv layer

### LINKED ISSUE #5164: need dilation support for exconv layer
Dilation commonly used in segmentation.
PaddlePaddle now support dilation in cudnn conv but not exconv , so, we need to extend it.

### LINKED ISSUE #5242: cnn_output_size Calculation error using dilation in python api
The origin func in python api as follows:
```
def cnn_output_size(img_size, filter_size, padding, stride, caffe_mode):
    output = (2 * padding + img_size - filter_size) / float(stride)
    if caffe_mode:
        return 1 + int(math.floor(output))
    else:
        return 1 + int(math.ceil(output))
```
This doesn't take dilation into account.

### PR DESCRIPTION
**Pull request by NHZlX**
fix  #5164 
fix #5242

### CODE REVIEW

File: paddle/function/ConvOpTest.h [hunk 1]
```diff
@@ -79,45 +79,54 @@ void Convolution(const std::string& conv1,
...
-                if (padding >= filterSize) break;
+                for (size_t dilation : {1, 3}) {
+                  if (padding >= filterSize) break;
 
-                // NNPACK only supports stride = 1 if batchSize > 1
-                if ((conv1 == "NNPACKConv-CPU" || conv2 == "NNPACKConv-CPU") &&
-                    batchSize > 1 && stride > 1)
-                  break;
+                  if ((conv1 == "NaiveConv-CPU" || conv2 == "NaiveConv-CPU") &&
+                      dilation > 1)
```
Comment (hedaoyuan) [inline paddle/function/ConvOpTest.h]: The `NNPACKConv` also does not support dilation.

File: paddle/function/ConvOpTest.h [hunk 2]
```diff
@@ -79,45 +79,54 @@ void Convolution(const std::string& conv1,
             if (outputChannels < inputChannels) continue;
             for (size_t stride : {1, 2}) {
               for (size_t padding : {0, 1}) {
-                if (padding >= filterSize) break;
+                for (size_t dilation : {1, 3}) {
+                  if (padding >= filterSize) break;
```
Comment (hedaoyuan) [inline paddle/function/ConvOpTest.h]: Why here do not need some checking like in the 'Im2ColTest.cpp`?
`filterSize` multiplied by `dilation` may be greater than `inputSize`.
```
                  size_t filterSizeH = (filterHeight - 1) * dilation + 1;
                  size_t filterSizeW = (filterWidth - 1) * dilation + 1;
                  if (inputHeight <= filterSizeH || inputWidth <= filterSizeW)
                    break;
                  if (padding >= filterSizeH || padding >= filterSizeW) break;
                  size_t outputHeight =
                      (inputHeight - filterSizeH + 2 * padding) / stride + 1;
                  size_t outputWidth =
                      (inputWidth - filterSizeW + 2 * padding) / stride + 1;
```

Comment (NHZlX) [reply paddle/function/ConvOpTest.h]: I add the control to it,  the `inputSize + 2 * padding < filterS`.

File: paddle/function/ConvOpTest.h (hunk 1 above)
Comment (NHZlX) [reply paddle/function/ConvOpTest.h]: Done

**Pull request merged**